python/plugins/MetaSearch/dialogs/maindialog.py

```diff
@@ -119,6 +119,9 @@ def __init__(self, iface):
         self.disable_ssl_verification = self.settings.value(
             "/MetaSearch/disableSSL", False, bool
         )
+        self.log_debugging_messages = self.settings.value(
```

Comment:
Can we not use the new settings API ?

Reply:
I think we can consider updating to the new settings API in another PR (to update all settings workflow).

Reply:
Ok for other settings, but this one just appear with this PR, it seems right to use the new API directly.

Reply:
@troopa81, may you please guide me in correctly implementing the new settings API in this case? I failed to find proper guidelines or code samples from other Python core plugins: e.g. the [PyQGIS Developer Cook Book](https://docs.qgis.org/testing/en/docs/pyqgis_developer_cookbook/settings.html) only says to use QgsSettings().
Anyway, I'm also very happy if someone would like to directly propose code changes to this PR or to my branch https://github.com/agiudiceandrea/QGIS/tree/metasearch-logging-setting or if someone would propose a new PR to correctly implement it.

Reply:
I'd prefer we cover settings API implementation in a subsequent PR. @troopa81 can you point us in the right direction/examples for the new settings API?

Reply:
@agiudiceandrea 

Take a look about this commit https://github.com/qgis/QGIS/pull/59762/commits/b494fc1ec5cda285c5e6fb9e85d449627e8f9839

It's C++ but principle stay the same. You can easily grab the treeRoot from QgsSettingsTree, call createChildNode with your group (MetaSearch for instance) and then create QgsSettingsEntryBool for your setting that you'll use later to set and get the value.